geolocation.js

```diff
@@ -60,44 +49,59 @@ function mergePositionOptions (
   }
 }
 
-// internal
-function requestCurrentPosition (
-  onSuccess /* : (position: PositionLike) => any */,
-  onError /* : (error: PositionError) => any */,
-  options /* :? PositionOptionsLike */
-) {
-  var geolocation = getGeoLocation()
-  if (!geolocation) {
-    throw new Error('the current web engine does not support GeoLocation')
+var DRIVERS_PREFERENCE = [ 'W3C' ]
+
+var DRIVERS /* : { [id:string]: GeolocationDriver } */ = {
+
+  W3C: {
+    isAvailable: function () /* : boolean */ {
+      return !!(
+        typeof navigator !== 'undefined' &&
+        navigator.geolocation &&
+        navigator.geolocation.getCurrentPosition
+      )
+    },
+
+    getCurrentPosition: function (
+      onSuccess /* : (position: PositionLike) => any */,
+      onError /* : (error: PositionError) => any */,
+      options /* : PositionOptions */
+    ) /* : void */ {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        onSuccess(clonePosition(position))
+      }, onError, options)
+    }
   }
-  options = mergePositionOptions(options)
-  geolocation.getCurrentPosition(function (position) {
-    onSuccess(clonePosition(position))
-  }, onError, options)
+
+}
+
+function detectDriver () /* : GeolocationDriver | false */ {
+  const availableDrivers = DRIVERS_PREFERENCE.filter((name) => {
+    return DRIVERS[name].isAvailable()
+  })
+  return DRIVERS[availableDrivers[0]] || false
```

Comment:
I did a fair bit of fence sitting between ES5 and ES2015 features. I should just commit to ES2015 and be done with it. I'm already using `const` here, so I may as well use `Array#find()`.